File: cityapi/src/main/kotlin/com/example/demo/entities/Ranch.kt
package com.example.demo.entities

import org.hibernate.annotations.GenericGenerator
import javax.persistence.*

@Entity
@Table(name = "ranches")
data class Ranch(
    @Id
    @GenericGenerator(name = "uuid2", strategy = "uuid2")
    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "uuid2")
    var id: String? ,
    val name: String,
    var description: String?,
    @ManyToOne
    @JoinColumn(name = "city_id")
    var city: City?,
    )

File: hello-word/src/types/ranch.ts
import { City } from "./City"

export class Ranch {
    id: String
    private name: String
    private description : String
    city: City | undefined

    constructor(id: String, name: String, description: String) {
        this.id = id
        this.name = name
        this.description = description
    }
}